core/src/network/actions/initialize_network.rs

```diff
@@ -20,14 +21,18 @@ pub async fn initialize_network(context: &Arc<Context>) -> HcResult<()> {
     let network_settings = NetworkSettings {
         config: context.network_config.clone(),
         dna_address,
-        agent_id,
+        agent_id: agent_id.clone(),
     };
     let action_wrapper = ActionWrapper::new(Action::InitNetwork(network_settings));
     dispatch_action(context.action_channel(), action_wrapper.clone());
 
     await!(InitNetworkFuture {
         context: context.clone(),
-    })
+    })?;
+
+    await!(publish(agent_id.clone().into(), context))?;
```

Comment:
Hm, yeah, I can see how that fixes this. 

And this also raises some question for me, that I would like to discuss with @neonphog:
This has to be a temporary hack, right? A real DHT and world model implementation would handle this intrinsically, right?

I guess the problem is that we first initialize the chain and then the network, so the committing of the agent_id happens before the network can get notified of this entry.

BUT: in the reducer of above `Action::InitNetwork` we register with the network with our agent_id. Shouldn't that be enough? ...and jumping a few months into the future: what about membranes? creating and publishing our agent_id is something that should also be validated eventually...